unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/AvatarSlot/AvatarSlotsHUDController.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DCL.Backpack
+{
+    public class AvatarSlotsHUDController : IHUD
+    {
+        internal AvatarSlotsDefinitionSO avatarSlotsDefinition;
+        internal readonly IAvatarSlotsView avatarSlotsView;
+        internal string lastSelectedSlot;
+
+        public AvatarSlotsHUDController(IAvatarSlotsView avatarSlotsView)
+        {
+            this.avatarSlotsView = avatarSlotsView;
+            avatarSlotsView.OnToggleAvatarSlot += ToggleSlot;
+            avatarSlotsDefinition = Resources.Load<AvatarSlotsDefinitionSO>("AvatarSlotsDefinition");
+        }
+
+        public void GenerateSlots()
+        {
+            for (var i = 0; i < avatarSlotsDefinition.slotsDefinition.Length; i++)
+            {
+                SerializableKeyValuePair<string, List<string>> section = avatarSlotsDefinition.slotsDefinition[i];
+                avatarSlotsView.CreateAvatarSlotSection(section.key, i < avatarSlotsDefinition.slotsDefinition.Length - 1);
+                foreach (string avatarSlotSection in section.value)
+                    avatarSlotsView.AddSlotToSection(section.key, avatarSlotSection);
+            }
+            avatarSlotsView.RebuildLayout();
```

Comment:
Is this completely needed? We should try to avoid as much as possible to force the layouts rebuildings.

Reply:
Sadly it is needed because when adding objects as children of a vertical layout it doesn't refresh automatically

Reply:
Ok, understood... although I think it would be a good practice to abstract the logic. It would be an easy change. Anyway, I agree with this option if we all decide to follow it 👍 Wdyt @lorux0?